### PR TITLE
feat(ClientApplication): v13 add role connections

### PR DESCRIPTION
**Pull request by jaw0r3k**
**Please describe the changes this PR makes and why it should be merged:**
 Backports #8855
**Status and versioning classification:**

### CODE REVIEW

File: typings/rawDataTypes.d.ts [hunk 1]
```diff
@@ -268,3 +270,12 @@ export interface APIAutoModerationRuleTriggerMetadata {
   regex_patterns?: string[];
   mention_total_limit?: number;
 }
+
+export interface APIApplicationRoleConnectionMetadata {
+	type: ApplicationRoleConnectionMetadataTypes;
+	key: string;
+	name: string;
+	name_localizations?: LocalizationMap;
+	description: string;
```
Comment (kyranet) [inline typings/rawDataTypes.d.ts:279]:
```suggestion
  type: ApplicationRoleConnectionMetadataTypes;
  key: string;
  name: string;
  name_localizations?: LocalizationMap;
  description: string;
```

**Review by kyranet — CHANGES_REQUESTED** [review]
Checked enums and compared to the referenced PR, all good, just one thing:

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

File: src/util/Constants.js [hunk 2]
```diff
@@ -1483,6 +1514,7 @@ function createEnum(keys) {
  * The type of an {@link ApplicationCommandPermissions} object.
  * @property {Object<ApplicationCommandType, number>} ApplicationCommandTypes
  * The type of an {@link ApplicationCommand} object.
+ * @property {Object{ApplicationRoleConnectionMetadataType, number}} ApplicationRoleConnectionMetadataTypes
```
Comment (Jiralite) [inline src/util/Constants.js:1517]: This is malformed.

**Review by Jiralite — CHANGES_REQUESTED** [review]

File: src/util/Constants.js (hunk 2 above)
Comment (SpaceEEC) [reply src/util/Constants.js]: Resolved with 360c2e861b84e24caf1c2e18f0a251168cdd84e6

**Pull request merged**